### DISCUSSION TITLE
Proposal - workgroups integration

### DISCUSSION THREAD

**Discussion by fvanderbiest** [georchestra/georchestra#4648] [Ideas]
Not yet an Improvement Proposal (this proposal is not funded at the time of writing).

GeoNetwork groups are currently synced either from:

- orgs - the default setting with geOrchestra
- roles - (older default) - still possible by tweaking the configuration / datadir

Both settings have pros and cons.

Syncing groups with orgs brings:
`+` simplicity: one user belongs to one org = one group to manage metadata
`+` readability: group name = full org name
`-` impossibility to have a user belong to 2+ geonetwork groups

Syncing groups with roles brings:
`+` possibility to have one user belong to 2+ geonetwork groups
`-` bad readability: group name = capitalized role name without its prefix (underscore being the separator)
`-` complexity for newcomers

We want to foster collaboration, and, as such, geOrchestra's default setting is not adequate anymore.

**We probably need an new "object kind" which maps to a "workgroup".**

Technically:

- workgroups would be stored in the LDAP as groupOfMembers (or apparented)
- workgroups would have members belonging to any organisation
- workgroups would be synced with geonetwork groups (potentially with geoserver namespaces ?)
- workgroups would be managed in the console

From this point on, there is a lot to discuss, but most importantly, would this bring real value or more complexity ?
Happy to read your thoughts on this.

See also https://groups.google.com/g/georchestra/c/PT0j-UC1gMA/m/oRxBxXEFBgAJ

**Comment by landryb** [comment]
my first thought reading this was *oh $deity, ppl already have a hard time grasping the difference between roles and orgs and misuse them most of the time, why adding another layer on top ?*

but then i pondered "hmm why not", but to me the concept of workgroup more or less matches the existing concept of roles..

how are ppl actually using geonetwork groups for, and what's the limitations of the current uses ?  editing of shared metadata ? giving access to resources to certain groups ? what else ?

why not bringing the concept of orgs directly into geonetwork ? after all, iirc geoserver knows both roles and orgs (unless i'm mistaken)

**Comment by landryb** [comment]
"notre usage des groupes GeoNetwork dans geOrchestra ne permet pas à un utilisateur de collaborer sur des métadonnées avec un autre issu d'un autre organisme."

isnt it already possible to give editing rights to other groups on a metadata belonging to a group ?

![image](https://user-images.githubusercontent.com/3421760/233053621-7bbc2290-c603-428f-b579-936d1a24b564.png)

(sure, one can't say "only this user from that other group" but..)

**Comment by fvanderbiest** [comment]
MEL says "Intéressant, Indispensable" !
